File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        isbalance = True
        def dfs(root):
            nonlocal isbalance
            if root is None:
                return 0
            lh = dfs(root.left)
            rh = dfs(root.right)

            if not -1<=(lh-rh)<=1:
                isbalance = False
            return 1+max(lh,rh)
        dfs(root)
        return isbalance



        

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0;
        int right = s.length() - 1;
        while (right > left) {
            if (isalnum(s[left]) && isalnum(s[right])) {
                if ((tolower(s[left])) != (tolower(s[right]))) {
                    return false;
                }
                left++;
                right--;
            }
            if (!isalnum(s[left])) {
                left++;
            }
            if (!isalnum(s[right])) {
                right--;
            }
        }
        return true;
    }
};